test: Remove vestigial top-level test

kriskowal opened the pull request:
We have found a test that appears at the top level and is no longer needed.

dckc reviewed (changes requested):
oops! on review, I see this is actually a `bundleSource` test for when I added the option of passing in filesystem access eplicitly. It should be moved under `packages/bundle-source`.

kriskowal commented:
It appears to be insufficient just to move the test file since it’s not driven by ava. I’m not sure whether I can convert this to an ava test and preserve the intent. I’ve posted an update regardless. I removed the harden assertion since the dir1 bundle doesn’t import harden anymore. I converted all of the CommonJS idioms to ESM idioms except for `__dirname`, which is dependent on the particulars of -r esm. I removed the “if executed as main” idiom because it is CommonJS specific and not compatible with the ava executor.

dckc reviewed (approved):
thanks for cleaning this up
